Default math values

AtlantiaKing opened the pull request:
Added default values for Vector3 (one, zero, right, up, forward), Vector2 (one, zero, right, up), Quaternion (identity) and Matrix (identity 4x4 and identity 3x3)

AtlantiaKing committed:
Added default vector, quaterion & matrix values

Added Matrix class with identity values for glm::mat4x4 and glm::mat3x3
Added Vector2 class with default values for glm::vec2
Added Vector3 class with default values for glm::vec3
Added identity value to the Quaternion class
DirectionalLight, DirectXMeshRenderer, BoxCollider, CanvasComponent, RectTransform and the Transform class now use these default math values.

AtlantiaKing committed:
Replaced more hardcoded values with default math values

Replaced hardcoded vector values in DirectionalLight, Transform, Rigidbody and PhysXObject classes

AtlantiaKing committed:
Fixed compile error in Matrix class

LeeVangraefschepe commented on Graphics/DirectionalLight.cpp:
```diff
@@ -3,14 +3,16 @@
+
 leap::graphics::DirectionalLight::DirectionalLight()
 {
 	SetTransform(
```
Empty transform should also be a default math value.

LeeVangraefschepe commented on Utils/Vector3.h:
```diff
@@ -0,0 +1,40 @@
+		inline static glm::vec3 m_One{ 1.0f, 1.0f, 1.0f };
+		inline static glm::vec3 m_Right{ 1.0f, 0.0f, 0.0f };
+		inline static glm::vec3 m_Up{ 0.0f, 1.0f, 0.0f };
+		inline static glm::vec3 m_Forward{ 0.0f, 0.0f, 1.0f };
```
Add the following default vectors
Back (0,0,-1)
Left (-1,0,0)
Down (0,-1,0)

LeeVangraefschepe commented on Utils/Vector2.h:
```diff
@@ -0,0 +1,33 @@
+		inline static glm::vec2 m_Zero{};
+		inline static glm::vec2 m_One{ 1.0f, 1.0f };
+		inline static glm::vec2 m_Right{ 1.0f, 0.0f };
+		inline static glm::vec2 m_Up{ 0.0f, 1.0f };
```
Add the following vector2s
Down (0,-1)
Left (-1,0)

LeeVangraefschepe reviewed (changes requested):
Check comments for more default values but overal great work.

AtlantiaKing committed:
Added identity 4x3 matrix

The DirectionalLight now uses identity 4x3 matrix instead of creating one itself using default vector3 values

AtlantiaKing committed:
Added additional values for Vector2 and Vector3

Added Left, Down and Back to the Vector3 class
Added Left and Down to the Vector2 class

LeeVangraefschepe reviewed (approved):
Feedback has been applied
